benchmarks/benchmark.ts

```diff
@@ -0,0 +1,17 @@
+import Simulation from '../src/Simulation'
+import { grid1, grid2 } from './levels'
+
+console.log('---Running benchmarks---')
+console.time('100 frames')
+let sim = new Simulation(grid1)
+let laserIndicator = sim.generateLaserIndicator()
+sim.initializeFromIndicator(laserIndicator)
+sim.generateFrames(100)
+console.timeEnd('100 frames')
+
+console.time('Hall of mirrors')
+sim = new Simulation(grid2)
+laserIndicator = sim.generateLaserIndicator()
+sim.initializeFromIndicator(laserIndicator)
+sim.generateFrames(100)
+console.timeEnd('Hall of mirrors')
```

Comment:
Running stuff once in JS is a not a good measure of performance, as the compiler is being busy recompiling the hot paths in background, and many functions are not yet flagged for optimization, but would be after a few more runs.

To do this properly, the easiest way is to use an benchmarking framework. I see other bench suite (testMul) is using "benchmark" package, why have you decided to not use it here?

Also I don't have any favourites here, but if you are looking for a change, then https://github.com/caderek/benny seems like a nice typescript layer on top of "benchmark" with great defaults.

Reply:
It was to show two different benchmarking options: a "naive" one and the other using the BenchmarkJS framework.
BenchmarkJS seems to be industry standard with 4k stars compared to benny but wasn't updated for some time. Both have very similar syntax.

Reply:
It feels the benchmarkJS / benny packages are more tailored to test function implementations and not full simulation runs, the output Event is:
- Run 1st benchmark level x 0.77 ops/sec ±5.28% (6 runs sampled)

Reply:
Benny is just a wrapper over BenchmarkJS. Use whichever you like more.

The output is indeed a bit cryptic once you get into such a long running functions like our simulation, but the information is there, and I believe the format is configurable.
If you are able to change the formatting, I would prefer it like:

- Hall of mirrors: 1298.70ms ±68.57ms (6 runs sampled)

That way we see the raw linear speed, not the inverse of it